mkosi/__init__.py

```diff
@@ -2295,6 +2303,38 @@ def calculate_signature(context: Context) -> None:
         )
 
 
+def calculate_signature_sop(context: Context) -> None:
+    pgptool = context.config.openpgp_tool
+    if context.config.key is None:
+        die("Signing key is mandatory when using SOP signing")
+
+    with (
+        complete_step("Signing SHA256SUMS…"),
+        open(context.staging / context.config.output_checksum, "rb") as i,
+        open(context.staging / context.config.output_signature, "wb") as o,
+    ):
+        run(
+            [pgptool, "sign", "/signing-key.pgp"],
+            env=context.config.environment,
+            stdin=i,
+            stdout=o,
+            sandbox=context.sandbox(
+                binary=pgptool,
+                options=[
+                    "--bind",
+                    context.config.key,
+                    "/signing-key.pgp",
```

Comment:
Please keep options and their arguments on the same line and disable auto formatting for this block, see rest of the code for examples